components/LessonCard.test.js

```diff
@@ -47,10 +52,25 @@ describe('Lesson Card Complete State', () => {
       }
     })
 
-    const { container } = render(
-      <LessonCard {...props} currentState="completed" />
-    )
-    expect(container).toMatchSnapshot()
+    render(<LessonCard {...props} currentState="completed" />)
+
+    expect(
+      screen.getAllByTestId('review-submissions-count')[0]
+    ).toBeInTheDocument()
+  })
+
+  test('Should render lessonCard with no submission count', async () => {
+    useQuery.mockReturnValue({
+      data: {
+        submissions: []
+      }
+    })
+
+    render(<LessonCard {...props} currentState="completed" />)
+
+    expect(
+      screen.queryByTestId('review-submissions-count')
+    ).not.toBeInTheDocument()
```

Comment:
What happens if there are no elements with data-testid="review-submissions-count"? Does screen.queryByTestId('review-submissions-count') return undefined and then it checks if undefined is not in the document like this: `expect(undefined).not.toBeInTheDocument()`?

Reply:
Yes exactly! If there no element found, `queryByTestId ` will return `undefined`. I could've done `expect(getByTestId(...))` but I think `queryByTestId` + `.not.toBeInDocument` is more descriptive.